sharkfuser/samples/pointwise/pointwise_ops.cpp

```diff
@@ -0,0 +1,266 @@
+// Copyright 2025 Advanced Micro Devices, Inc.
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <fusilli.h>
+
+#include "utils.h"
+
+#include <algorithm>
+#include <catch2/catch_test_macros.hpp>
+#include <catch2/generators/catch_generators.hpp>
+#include <cstdint>
+#include <memory>
+#include <tuple>
+#include <unordered_map>
+#include <vector>
+
+using namespace fusilli;
+
+namespace {
+
+// Helper to allocate Buffer filled by initial value
+inline std::shared_ptr<Buffer>
+allocateBufferOfType(Handle &handle, const std::shared_ptr<TensorAttr> &tensor,
```

Comment:
There exists a function of this name in `tests/utils.h` which is packaged as a standalone utils library (that samples depend on via `libutils`). I'd avoid duplicating another function here and instead reuse/extend that one . The signature is slightly different, I think we can stick to what you have here (take in a tensor instead of shape + volume).